store/tikv/client.go

```diff
@@ -45,105 +55,230 @@ const (
 	readTimeoutShort  = 20 * time.Second  // For requests that read/write several key-values.
 	readTimeoutMedium = 60 * time.Second  // For requests that may need scan region.
 	readTimeoutLong   = 150 * time.Second // For requests that may need scan region multiple times.
+
+	rpcLabelKV  = "kv"
+	rpcLabelCop = "cop"
 )
 
 type rpcClient struct {
-	msgID uint64
-	p     *Pools
+	p *Pools
 }
 
 func newRPCClient() *rpcClient {
 	return &rpcClient{
-		msgID: 0,
 		p: NewPools(maxConnection, func(addr string) (*Conn, error) {
 			return NewConnection(addr, dialTimeout)
 		}),
 	}
 }
 
-// SendCopReq sends a Request to co-processor and receives Response.
-func (c *rpcClient) SendCopReq(ctx goctx.Context, addr string, req *coprocessor.Request, timeout time.Duration) (*coprocessor.Response, error) {
-	start := time.Now()
-	defer func() { sendReqHistogram.WithLabelValues("cop").Observe(time.Since(start).Seconds()) }()
-
-	select {
-	case <-ctx.Done():
-		return nil, ctx.Err()
-	default:
-	}
+func (c *rpcClient) callFunc(addr string, f func(conn *Conn) error, label string) error {
```

Comment:
You can accept `f func(TiKVClient) error` then you don't have to `client := tikvpb.NewTiKVClient(conn.ClientConn)` in every `f`.